app/src/main/java/com/rockthevote/grommet/ui/eventFlow/SessionTimeTrackingViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package com.rockthevote.grommet.ui.eventFlow
+
+import androidx.lifecycle.*
+import com.hadilq.liveevent.LiveEvent
+import com.rockthevote.grommet.data.db.dao.PartnerInfoDao
+import com.rockthevote.grommet.data.db.dao.SessionDao
+import com.rockthevote.grommet.util.coroutines.DispatcherProvider
+import com.rockthevote.grommet.util.coroutines.DispatcherProviderImpl
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+/**
+ * Created by Mechanical Man on 5/30/20.
+ */
+class SessionTimeTrackingViewModel(
+        private val dispatchers: DispatcherProvider = DispatcherProviderImpl(),
+        private val partnerInfoDao: PartnerInfoDao,
+        private val sessionDao: SessionDao
+) : ViewModel() {
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+        Timber.e(throwable)
+    }
+
+    private val _effect = LiveEvent<SessionSummaryState.Effect?>()
+    val effect: LiveData<SessionSummaryState.Effect?> = _effect
+
+    val sessionData = Transformations.map(partnerInfoDao.getPartnerInfoWithSessionAndRegistrations()) { result ->
+        result?.let{
+            val partnerInfo = result.partnerInfo
+            val session = result.sessionWithRegistrations?.session
+            val registrations = result.sessionWithRegistrations?.registrations
+
+            SessionSummaryData(
+                    partnerInfo?.partnerName ?: "",
+                    session?.canvasserName ?: "",
+                    session?.openTrackingId ?: "",
+                    session?.partnerTrackingId ?: "",
+                    session?.deviceId ?: "",
+                     session?.smsCount ?: 0,
+                    session?.driversLicenseCount ?: 0,
+                    session?.ssnCount ?:0,
+                    session?.emailCount ?: 0,
+                    session?.registrationCount ?: 0,
+                    session?.abandonedCount ?: 0,
+                    registrations ?: emptyList(),
+                    session?.clockInTime
+
+            )
+        } ?: run {
+            SessionSummaryData()
+        }
+    }
+
+    fun clearSession() {
+        viewModelScope.launch(dispatchers.io + coroutineExceptionHandler) {
+            sessionDao.clearAllSessionInfo()
+            // TODO listen for return value and set error state if it fails?
```

Comment:
calling out this comment

Reply:
I don't think the chance of this happening is severe enough to where I'd normally worry about that.

That said, and more for informational purposes. If `clearAllSessionInfo` does error out it'll be in the form of an exception. Because this is wrapped in a `launch` command on a coroutine with a `coroutineExceptionHandler`, the exception, if it occurs in `clearAllSessionInfo` would terminate this coroutine and propagate the exception to the exception handler while the app crashes.

Reply:
Correction, the exception handler will not let the app crash, it will hang. I'm fixing this in another PR right now though.